src/mirage_kv.mli

```diff
@@ -207,6 +225,11 @@ module type RW = sig
      enclosing {!batch} operation, where durability will be guaranteed
      at the end of the batch. *)
 
+  val rename: t -> key -> key -> (unit, write_error) result Lwt.t
```

Comment:
It'd probably be good to make explicit that dictionaries can be replaced by keys, which some users might find surprising.

Reply:
I'd label the arguments: `rename ~old ~new` (likely there are better labels). What is the intended semantics if `new` already exists? What if `old` does not exist? These edge cases should be documented.